docker/Dockerfile

```diff
@@ -0,0 +1,130 @@
+# Handle ros distro
+ARG ROS_DISTRO=noetic
+FROM ghcr.io/aica-technology/ros-ws:${ROS_DISTRO}
+
+# User provided arguments
+ARG HOST_GID=1000
+ARG GIT_NAME=""
+ARG GIT_EMAIL=""
+ARG USE_SIMD=OFF
+
+# Tell docker we want to use bash instead of sh in general
+SHELL ["/bin/bash", "-c"]
+
+### Add the user to the current GID of the host to avoid permisson issues in volumes
+# AICA uses the same name for user and user group
+ENV USER_GROUP=${USER}
+USER root
+RUN if [ "HOST_GID" != "1000"] ; \
+    then groupadd --gid ${HOST_GID} host_group && \
+    usermod ${USER} -g ${HOST_GID} && \ 
+    usermod ${USER} -a -G ${USER_GROUP}; fi
+USER ${USER}
+
+# Setup git identity
+RUN git config --global user.name "${GIT_NAME}"
+RUN git config --global user.email "${GIT_EMAIL}"
+
+# Setup python version for noetic
+RUN sudo apt update
+RUN if [ "${ROS_DISTRO}" == "noetic" ] ; \
+    then sudo apt install python-is-python3 ; fi
+
+
+### Add a few tools
+RUN sudo apt-get update && sudo apt-get install -y \
+    bash-completion \
+    silversearcher-ag \
+    apt-transport-https \
+    less \
+    alsa-utils \
+    ros-${ROS_DISTRO}-ros-control \
+    ros-${ROS_DISTRO}-ros-controllers \
+    && sudo apt-get upgrade -y && sudo apt-get clean
+
+# Install gazebo (9 or 11 depending on distro)
+WORKDIR /home/${USER}
+RUN sudo apt-get update
+RUN if [ "$ROS_DISTRO" = "noetic" ] ; then sudo apt-get install -y gazebo11 ; fi
+RUN if [ "$ROS_DISTRO" = "melodic" ] ; then sudo apt-get install -y gazebo9 ; fi
+
+# Install gaezbo ros packages
+RUN sudo apt install -y ros-${ROS_DISTRO}-gazebo-ros-pkgs ros-${ROS_DISTRO}-gazebo-ros-control
+
+# Handle SIMD option
+RUN if [ "${USE_SIMD}" = "ON" ] ; \ 
+    then export CMAKE_CXX_FLAGS="-march=native -faligned-new" ; fi
+
+### Install all dependencies of IIWA ROS
+# Clone KUKA FRI (need to be root to clone private repo)
+WORKDIR /tmp
+USER root
+RUN --mount=type=ssh git clone git@github.com:epfl-lasa/kuka_fri.git
```

Comment:
This would fail for everyone not having access (like me)
But I guess people can adapt the docker file to make it point to their own repo. Or pass and argument maybe?

Reply:
Oh it was my understanding that this was a requirement to use this library. If that's not the case I can find a work around to skip it in case we can't access

Reply:
It is required, you got that totally right. But I am not at EPFL and haven't gotten access to the repo.
I got the code instead and put it in our own repo. So for me the docker would not build right now.

I guess one option is to allow people to optionally pass their own repo address

Reply:
This is a good point by @matthias-mayr . I want to make `kuka_fri` public though in the long term. I need to contact KUKA. So for now, we can make this an argument..